flooder/src/main.rs

```diff
@@ -1,100 +1,258 @@
-
-                    info!("sending #{} transactions", tx_count);
-                    for _ in 0..tx_count {
-                        conn.transfer_keep_alive(dest.clone(), transfer_amount, status)
-                            .await
-                            .unwrap();
-                    }
+    let start = Instant::now();
+    let mut nonces: Vec<u32> = vec![];
+    for conn in &connections {
+        nonces.push(conn.account_nonce(conn.account_id()).await?);
+    }
 
-                    let dur = time::Instant::now().saturating_duration_since(start);
+    let mut overdue_transactions = 0;
+    for i in 0..schedule.intervals {
+        let pending_in_pool = connections[0].pending_extrinsics_len().await?;
+        overdue_transactions += schedule.transactions_in_interval;
+        let transactions_to_pool_limit = pool_limit.saturating_sub(pending_in_pool);
+        let transactions_to_send = min(transactions_to_pool_limit, overdue_transactions);
+        overdue_transactions -= transactions_to_send;
+        info!(
+            "Interval {}: submitting {} txns",
+            i + 1,
+            transactions_to_send
+        );
+        debug!("In txpool there are {pending_in_pool} txns. Overdue txns: {overdue_transactions}");
 
-                    let left_duration = Duration::from_secs(time).saturating_sub(dur);
+        let tasks = connections
+            .iter()
+            .enumerate()
+            .map(|(conn_id, conn)| {
+                let mut per_task = transactions_to_send / connections.len() as u64;
+                if (conn_id as u64) < transactions_to_send % connections.len() as u64 {
+                    per_task += 1;
+                }
 
-                    info!("sleeping for {}ms", left_duration.as_millis());
-                    sleep(left_duration).await;
+                let dest = dest.clone();
+                let nonce = nonces[conn_id];
+                nonces[conn_id] += per_task as u32;
+                async move {
+                    for tx_id in 0..per_task as u32 {
+                        conn.sign_with_params(
+                            transfer_keep_alive(dest.clone(), transfer_amount),
+                            Default::default(),
+                            nonce + tx_id,
+                        )?
+                        .submit(status)
+                        .await?;
+                    }
+                    anyhow::Ok(())
                 }
             })
-        })
-        .collect();
+            .collect::<Vec<_>>();
 
-    join_all(handles).await;
+        for result in join_all(tasks).await {
+            if let Err(e) = result {
+                info!("Error submitting transaction: {e:?}");
+                return Err(e);
+            }
+        }
+
+        let duration = Instant::now().saturating_duration_since(start);
+        if duration.as_secs() >= schedule.interval_duration * schedule.intervals {
+            break;
+        }
```

Comment:
transactions now are not sent in intervals. Setting interval_duration to 1sec does not mean all planned intervals will last below 1sec. It rather means that first interval that went over 1sec ends the whole flooding.

Reply:
The timer runs from the start, so this checks the total flooding duration. Intervals try to last 1sec, but if previous ones ended too late (can happen mostly because rpc node can't confirm adding that many of them to the pool in that short time), those intervals will "try to catch up". So finally, there could be less intervals than `schedule.intervals` in such situations.

Reply:
Yep, this is confusing, lest fix it. Duration interval should be enforced and then the number of intervals also should be enforced. what about timeout set to duration interval?

Reply:
Kinda fixed, is this good enough?

Reply:
Actually, I reverted this and refactored to use `time::interval` await with `MissedTickBehavior::Burst`. With this refactor, I see no point in artificially enforcing constant interval lengths, especially since changes caused by spawning tasks made distributing overdue transactions less balanced. Is it ok for you?

Reply:
ok